WordPress/openverse-frontend
# Convert `usage-data` store to an options Pinia store

## Fixes


Fixes #1026 by @obulat

## Description


This PR converts `usage-data` store to Pinia. This store handles sending of the analytics data points to the API, and doesn't contain any state, so it might be a good idea to refactor it into a composable in the future.
For now, I converted it from Vuex to Pinia so that we can keep converting other stores that depend on it. 

This store needs to access `$sentryReady` from the Nuxt context. Because the Pinia options store has  the context available on `this.$nuxt`, I used an `options` store (with actions, similar to Vuex), and not the `setup` store (similar to composable functions).

## Testing Instructions

Enable Sentry and run the app: `ENABLE_INTERNAL_ANALYTICS=true pnpm run dev`

Revert the fix for analytics endpoint API bug that results in double slashes at the end:
https://github.com/WordPress/openverse-frontend/blob/7d886d34eb84a76cf8e8c9c58e0b319b6dc0b348/src/data/usage-data-service.js#L17
This will cause analytics requests to throw a network error. When searching, opening the detail page, clicking on the 'Copy attribution', you should see the sentry post requests in the console. You will have to disable ad blocking extensions if you have any.
The unit tests have almost 100% coverage (the internal of Sentry are not covered).

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
